providers/file_input.rb

```diff
@@ -28,6 +30,8 @@
               'state_file' => new_resource.name,
               'severity' => new_resource.severity,
               'facility' => new_resource.facility
-    notifies :restart, resources(:service => 'rsyslog')
+    notifies :restart, resources('service[rsyslog]')
   end
+
+  new_resource.updated_by_last_action(t.updated_by_last_action?)
```

Comment:
Gotcha, I got it. So do we need to duplicate the same behaviour that's in the `default` recipe ?

```ruby
if node['platform'] == 'ubuntu' && node['platform_version'].to_f >= 12.04
  service_provider = Chef::Provider::Service::Upstart
else
  service_provider = nil
end

service node['rsyslog']['service_name'] do
  supports :restart => true, :status => true
  action   [:enable, :start]
  provider service_provider
end
```